specs/RasterizationScale.md

```diff
@@ -0,0 +1,164 @@
+            std::to_wstring(int(rasterizationScale * 100)) + L"%)";
+        SetWindowText(m_appWindow->GetMainWindow(), message.c_str());
+        return S_OK;
+    })
+.Get(), &m_rasterizationScaleChangedToken));
+```
+## .Net
+```c#
+public void SetRasterizationScale(double additionalScale)
+{
+    CoreWebView2Controller.ShouldDetectMonitorScaleChanges = false;
+    m_webviewAdditionalRasterizationScale = additionalScale;
+    
+    // RasterizationScale is typically the monitor DPI scale and text scaling, but
+    // the app could add an additional scale for its own scenarios.
+    double rasterizationScale = additionalScale * GetDpiScale() * GetTextScale();
+    CoreWebView2Controller.RasterizationScale = rasterizationScale;
+}
+```
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+```c#
+interface ICoreWebView2Controller2 : ICoreWebView2Controller {
+  /// The rasterization scale for the WebView. The rasterization scale is the
+  /// combination of the monitor DPI scale and text scaling set by the user.
+  /// This value should be updated when the DPI scale of the app's top level
+  /// window changes (i.e. monitor DPI scale changes or window changes monitor)
+  /// or when the text scale factor of the system changes.
+  ///
+  /// \snippet AppWindow.cpp DPIChanged
+  ///
+  /// \snippet AppWindow.cpp TextScaleChanged1
+  ///
+  /// \snippet AppWindow.cpp TextScaleChanged2
+  ///
+  /// Rasterization scale applies to the WebView content, as well as
+  /// popups, context menus, scroll bars, and so on. Normal app scaling scenarios
+  /// should use the ZoomFactor property or SetBoundsAndZoomFactor API which
+  /// only scale the rendered HTML content and not popups, context menus, scroll bars and so on.
+  ///
+  /// \snippet ViewComponent.cpp RasterizationScale
+  [propget] HRESULT RasterizationScale([out, retval] double* scale);
+  // Set the rasteriation scale property.
+  [propput] HRESULT RasterizationScale([in] double scale);
+
+  /// ShouldDetectMonitorScaleChanges property determines whether the WebView
+  /// attempts to track monitor DPI scale changes. When true, the WebView will
+  /// track monitor DPI scale changes, update the RasterizationScale property,
+  /// and raises RasterizationScaleChanged event. Attempting to set RasterizationScale
+  /// while ShouldDetectMonitorScaleChanges is true will result in RasterizationScaleChanged
+  /// being raised and the value restored to match the monitor DPI scale.
+  /// When false, the WebView will not track monitor DPI scale changes, and the app
+  /// must update the RasterizationScale property itself. RasterizationScaleChanged
+  /// event will never raise when ShouldDetectMonitorScaleChanges is false.
+  [propget] HRESULT ShouldDetectMonitorScaleChanges([out, retval] BOOL* value);
+  /// Set the ShouldDetectMonitorScaleChanges property.
+  [propput] HRESULT ShouldDetectMonitorScaleChanges([in] BOOL value);
```

Comment:
Another option that (I think) aligns with XAML is to have two properties

```csharp
IReference<double> RasterizationScaleOverride; // default null
double RasterizationScale{get;};
```

If `RasterizationScaleOverride` is null, then no override is in effect, and the platform controls the rasterization scale. (Formerly, `ShouldDetectMonitorScaleChanges=true`.)

If `RasterizationScaleOverride` is non-null, then it is used, and the system does not perform automatic scale changs. (Formerly, `ShouldDetectMonitorScaleChanges=false`, `RasterizationScale=custom_scale`.)

Regardless, you can always query the current rasterization scale by reading `RasterizationScale`.

This gets rid of the weird interaction between the old `RasterizationScale` and the old `ShouldDetectMonitorScaleChangse`.

`IReference<double>` may be cumbersome for C++ clients, though.

You could do it the XAML way and make `RasterizationScaleOverride` a `double` and use `NaN` to mean "no override in effect."

Or you could let `-1` be the "no override" value, since a negative scale makes no sense. (A scale of 0 is still meaningful - it means everything shrinks to a point, and could be a boundary case in some extreme low DPI scenarios.)

Reply:
+1 to having a read/write "override" property an a read-only "actual" property.

The only place I see almost exactly this Override pattern is [Contact.DisplayNameOverride](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.Contacts.Contact.DisplayNameOverride).

Xaml uses "Actual" a lot for the read-only property, for example [FrameworkElement.Height](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.FrameworkElement.Height) and [FrameworkElement.ActualHeight](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.FrameworkElement.ActualHeight).

I don't like non-standard sentinels like double.nan and int32.min/max or -1. It's easier to remember that regardless of the type, the sentinel value is null.

I like best: nullable RasterizationScale and read-only ActualRasterizationScale.